test/signs_ui_web/controllers/messages_controller_test.exs

```diff
@@ -9,21 +9,30 @@ defmodule SignsUiWeb.MessagesControllerTest do
   }
 
   describe "index" do
+    @tag :authenticated
     test "lists all messages", %{conn: conn} do
-      conn =
-        conn
-        |> add_req_header(:basic)
-        |> get(messages_path(conn, :index))
+      conn = get(conn, messages_path(conn, :index))
 
       assert html_response(conn, 200) =~ "MBTA Realtime Signs"
     end
+
+    @tag :authenticated
+    test "includes user token", %{conn: conn} do
+      conn = get(conn, messages_path(conn, :index))
+
+      response = html_response(conn, 200)
+
+      token = Regex.run(~r/window.userToken = "(.+)";/, response) |> Enum.at(1)
```

Comment:
Should the token be available to Javascript? I thought the best practice here was to have the token as an HTTPOnly cookie, so that the browser is responsible for sending it but that it's not available to attackers?

Reply:
This is the approach [recommended by Chris McCord](https://elixirforum.com/t/accessing-cookies-in-phoenix-socket-connect/4731/4) in a question about using HTTPOnly cookies with websockets. Phoenix by design doesn't give access to the cookie in the socket/channel callbacks and recommends passing a signed `Phoenix.Token` via JS like we're doing. He says that it's more secure this way anyway, since websockets aren't subject to the same-origin cookie policy that http requests are.

Reply:
Thanks: I didn't know about that attack, so I learned something today!